SPEC.md

```diff
@@ -319,17 +319,92 @@ The maximum CPU speed is 100, setting any number higher is the same as setting
 to 100. The game operates at 10 frames per second meaning a fully overclocked
 CPU will execute 1000 instructions per second, 100 per frame.
 
+### Battery Check
+
+A robot can know how much battery it has:
+
+```
+push8 #IO_BATTERY
+io
+popf batteryLevel
+```
+
+The float on the stack will be between `0.0` and `1.0` describing how much
+battery is remaining.
+
 ### Laser
 
-*Not yet implemented*
+Robots get a laser beam which can damage some obstacles and other robots in
+combat. Currently a laser can break rocks which may be blocking doors. Having
+the laser on costs up to `256` battery per game frame, less if the laser is
+not fully powered or hits something early.
 
-Will deal battery damage to other bots.
+```
+; Full power laser
+pushf 1.0
+push8 #IO_LASER
+io
+
+; Laser off
+pushf 0.0
+push8 #IO_LASER
+io
+```
 
 ### Accelerometer
 
+The accelerometer allows a robot to detect relative changes in it's position.
+Each time you use the IO_ACCELEROMETER command the last position is saved and
+the difference is pushed on the stack as two floats.
+
 ```
 push8 #IO_ACCELEROMETER
 io
 popf relY
 popf relX
 ```
+
+### Compass Tracking
+
+The compass allows a robot to determine a general location in the game world.

```

Comment:
The returned value ranges from -PI to +PI and represents the angle from the center of the world. Being below the "equator" would return a value from 0.0 to +PI while being above would return a value of -PI to 0.0